### PR TITLE
Better document the SQL operators

### LINKED ISSUE #346: sql: Add special SQL functions to functions-and-operators.md
These [SQL functions](https://www.cockroachlabs.com/docs/sql-grammar.html#func_expr_common_subexpr) show up in the grammar. But they do not show up in our auto-generated [functions](https://www.cockroachlabs.com/docs/functions-and-operators.html) docs.

@mjibson and I decided the best approach is to manually add a section for these functions.

### PR DESCRIPTION
**Pull request by knz**
Fixes #346.

After we're OK with copy-edit I can propagate the changes to the v2.0 docs.

### CODE REVIEW

**Comment by mjibson** [comment]
This is great. Would have saved me a lot of time. Thanks.

File: v2.1/functions-and-operators.md [hunk 1]
```diff
@@ -4,12 +4,38 @@ summary: CockroachDB supports many built-in functions, aggregate functions, and
...
 
+## Special Syntax Forms
+
+The following syntax forms are recognized for compatibility with the
+SQL standard and PostgreSQL, but are equivalent to regular built-in
+functions:
+
+{% include sql/{{ page.version.version }}/function-special-forms.md %}
+
+## Conditional And Function-Like Operators
```
Comment (jseldess) [inline v2.1/functions-and-operators.md]: nit: `And` > `and`

File: v2.1/functions-and-operators.md [hunk 2]
```diff
@@ -4,12 +4,38 @@ summary: CockroachDB supports many built-in functions, aggregate functions, and
...
+
+The following table lists the operators that look like built-in
```
Comment (jseldess) [inline v2.1/functions-and-operators.md]: nit: end with colon instead of period.

File: v2.1/scalar-expressions.md [hunk 3]
```diff
@@ -670,6 +605,71 @@ Check our blog for
...
+( ... subquery ... )
+~~~
+
+Evaluates the subquery, asserts that it returns a single row and single column,
+and then evaluates to the value of that single cell. Any [selection query](selection-queries.html)
+can be used as subquery.
+
+For example:
+
+~~~sql
```
Comment (jseldess) [inline v2.1/scalar-expressions.md]: Would you mind added copy-to-clipboard functionality to this and all other command examples on this page?

**Comment by knz** [comment]
> I'm wondering if this PR also resolves #346, though I don't see ANY/SOME/ALL documented, so perhaps that's something different?

Yes it resolves #346. I just added ANY/SOME/ALL.

> A question distinct from this PR: Would it be possible to somehow auto-generate function-special-forms.md and the operators order of precedence table from source? I imagine that wouldn't be straight-forward, but it might be something to consider as a future project.

I have filed https://github.com/cockroachdb/cockroach/issues/26088 to track this.

---

Review status: 0 of 6 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[v2.1/functions-and-operators.md, line 21 at r1](https://reviewable.io/reviews/cockroachdb/docs/3192#-LDMLdf74w2pWudyHUNm-r1-21:-LDMdyOR-Cd8QuHTBFpd:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/66358b0cc179c725fa63dbf3d5bb996cb149ce51/v2.1/functions-and-operators.md#L21)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: `And` > `and`
</blockquote></details>

Done.

---

*[v2.1/functions-and-operators.md, line 23 at r1](https://reviewable.io/reviews/cockroachdb/docs/3192#-LDMLdf74w2pWudyHUNm-r1-23:-LDMdzZIBmS02FO8V2Ne:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/66358b0cc179c725fa63dbf3d5bb996cb149ce51/v2.1/functions-and-operators.md#L23)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: end with colon instead of period.
</blockquote></details>

Done.

---

*[v2.1/scalar-expressions.md, line 624 at r1](https://reviewable.io/reviews/cockroachdb/docs/3192#-LDMLdf74w2pWudyHUNn-r1-624:-LDMdHw23QR-ITTUAbdn:btzkk3n) ([raw file](https://github.com/cockroachdb/docs/blob/66358b0cc179c725fa63dbf3d5bb996cb149ce51/v2.1/scalar-expressions.md#L624)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Would you mind adding copy-to-clipboard functionality to this and all other command examples on this page?
</blockquote></details>

Honestly I'd rather not. It would be a lot of work and dilute the focus of this PR. Feel free to assign an issue to this effect to me for a later "docs fixitday".

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3192#-:-LDMe2Sl41AIx9TkNBMZ:b-rshs09)*

**Comment by knz** [comment]
I have propagated the changes to the v2.0 tree.

**Pull request merged**